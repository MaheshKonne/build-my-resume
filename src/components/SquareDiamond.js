import React from "react";
import styled from "styled-components";
import {theme} from "../lib/theme";

const SquareDiamond = styled.div`
    width: ${props => props.size}px;
    height: ${props => props.size}px;
    background: ${props => props.color};
    transform: rotate(45deg);
`;

const Diamond = ({size = 10, color = theme.white}) => {
    return (
        <SquareDiamond size={size} color={color} />
    );
};

export default Diamond;