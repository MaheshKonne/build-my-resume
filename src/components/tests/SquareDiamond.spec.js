import React from "react";
import renderer from "react-test-renderer";
import SquareDiamond from '../SquareDiamond';

describe('<SquareDiamond />', () => {
    it('should match with snapshot', () => {
        const component = renderer.create(<SquareDiamond />).toJSON();
        expect(component).toMatchSnapshot();
    });
});