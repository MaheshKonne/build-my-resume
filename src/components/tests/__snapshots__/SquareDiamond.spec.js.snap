// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SquareDiamond /> should match with snapshot 1`] = `
.c0 {
  width: 10px;
  height: 10px;
  background: #fff;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<div
  className="c0"
  color="#fff"
  size={10}
/>
`;
